lib/ret_web/channels/hub_channel.ex

```diff
@@ -66,7 +66,12 @@ defmodule RetWeb.HubChannel do
         _ -> nil
       end
 
-    hub_requires_oauth = hub.hub_bindings |> Enum.empty?() |> Kernel.not()
+    hub_requires_oauth =
```

Comment:
Feels slightly weird to be going through the rest of this function with hub being nil... It might burn us in the future. Might be better to instead match on `  defp perform_join(socket, nil, context, params) do` and just return an error right away (or immediately call `join_with_hub(nil,nil,nil,nil,nil)` if we want to keep a single codepath for error handling.

Reply:
Hmm, yeah, good point. I'll try a path that exits faster.